doc/user/flowspec.rst

```diff
@@ -0,0 +1,345 @@
+           pkts 1000, bytes 1000000
+        to 5.5.5.2:proto 17:50-90 (5)
+           pkts 1692918, bytes 157441374
+
+As you can see, the entry is present. note that an `iptable` entry can be used to host
+several Flowspec entries. In order to know where the matching traffic is redirected to,
+you have to look at the policy routing rules. The policy-routing is done by forwarding
+traffic to a routing table number. That routing table number is reached by using a
+`iptable`. The relationship between the routing table number and the incoming traffic
+is a MARKER that is set by the IPtable referencing the IPSet. In Flowspec case,
+`iptable` referencing the `ipset` context have the same name. So it is easy to know which
+routing table is used by issuing following command:
+
+.. code-block:: frr
+
+   CLI# show pbr iptable
+      IPtable match0x271ce00 action redirect (5)
+        pkts 1700000, bytes 158000000
+        table 257, fwmark 257
+   ...
+
+As you can see, by using following Linux commands, the MARKER `0x101` is present in
+both ``iptable`` and ``ip rule`` contexts.
+
+.. code-block:: shell
+
+   # iptables -t mangle --list match0x271ce00 -v
+   Chain match0x271ce00 (1 references)
+   pkts bytes target     prot opt in     out     source               destination
+   1700K  158M MARK       all  --  any    any     anywhere             anywhere
+        MARK set 0x101
+   1700K  158M ACCEPT     all  --  any    any     anywhere             anywhere
+
+   # ip rule list
+   0:from all lookup local
+   0:from all fwmark 0x101 lookup 257
+   32766:from all lookup main
+   32767:from all lookup default
+
+This allows us to see where the traffic is forwarded to.
+
+.. _flowspec-known-issues:
+
+Limitations / Known issues
+--------------------------
+
+As you can see, Flowspec is rich and can be very complex.
+As of today, not all flowspec rules will be able to be converted into Policy
+Based Routing actions.
+
+- The ``Netfilter`` driver is not integrated into FRR yet. Not having this piece
+  of code prevents from injecting flowspec entries into the underlying system.
+
+- There are some limitations around filtering contexts
+
+  If I take example of UDP ports, or TCP ports in flowspec, the information
+can be a range of ports, or a unique value. This case is handled.
+However, complexity can be increased, if the flow is a combination of a list of
+range of ports and an enumerate of unique values. Here this case is not handled.
+Similarly, it is not possible to create a filter for both src port and dst port.
+For instance, filter on src port from [1-1000] and dst port = 80.
+The same kind of complexity is not possible for packet length, ICMP type, ICMP code.
+
+- The first version of FRR Flowspec only handles a subset of Filter rules.
+
+  It operates policy based routing, on the following Flowspec criteria : IP Address,
+UDP port or TCP port. The next implementation plans to provide the following : support for
```

Comment:
Same here. Please see the Sphinx docs on how to write lists.

Reply:
ok